### ISSUE TITLE
Add tests for HTTP Range requests

### ISSUE THREAD

**Issue by lidel** [ipfs/gateway-conformance#154]
## TLDR

Coverage of range behaviors is bare minimum for UnixFS right now.
We should improve it and both relax requirements (allowing implementers to choose what to support) and make asserts more explicit (if you choose to implement something optional, you should behave A, if you don't, we expect fallback to be either B or C etc).

When it comes to range requests, HTTP specs ([RFC9110](https://httpwg.org/specs/rfc9110.html#range.requests)) make them OPTIONAL. Supporting HTTP Ranges is useful for streaming chunks of deserialized videos, and that is what some IPFS Gateways support when a client asks for deserialized data (a single range).

## Missing tests

This means these behaviors may occur:
- returned HTTP 206 with bytes for the first range
- returned HTTP 206 with content type `multipart/byteranges` and multiple ranges 
  - this is not used in practice, browsers streaming videos ask for a single range, Kubo >=0.23 does not support this for unixfs.
- returned HTTP 200 with the whole file 
- returned HTTP 200 with the whole file

a file could be any type of deserialized response, but in practice end users only care about UnixFS file

We don't want to be too strict here, but also we want to be able to write a test that limits the possibility space to the above paths.
If someone implements Ranges, we want to ensure the response is valid.
If they don't we also want to ensure the response is valid.

## What we need in DSL

It is most likely that we need to extend test DSL with something like `AnyOf` described in https://github.com/ipfs/gateway-conformance/issues/153

**Comment by lidel** [comment]
We have tests for trustless `?format=car&entity-bytes=0:0` but we are lacking a similar test for deserialized responses with HTTP header (`Range: bytes=0-0`).

- [ ] **HTTP Range request works and does overfetch**
  - Reuse [gateway-conformance/fixtures/trustless_gateway_car/file-3k-and-3-blocks-missing-block.car](https://github.com/ipfs/gateway-conformance/raw/ed74b532fc7662732ec87d92c49435668369db66/fixtures/trustless_gateway_car/file-3k-and-3-blocks-missing-block.car): A file chunked into 3 blocks, each 1024 bytes. The middle block is missing to ensure range request only triggers fetch of necessary blocks.
  - HTTP `Range: bytes=0-0` request for `/ipfs/QmYhmPjhFjYFyaoiuNzYv8WGavpSRDwdHWe5B4M5du5Rtk`, expect response to be HTTP 206 with    body equal to the first byte of the file
  - HTTP `Range: bytes=2200-` request for `/ipfs/QmYhmPjhFjYFyaoiuNzYv8WGavpSRDwdHWe5B4M5du5Rtk`, expect response to be HTTP 206 with   body equal to the expected tail of the file

**Comment by lidel** [comment]
Another thing that we should have test for: https://github.com/ipfs/boxo/issues/856
